pkg/render/policyrecommendation.go

```diff
@@ -0,0 +1,325 @@
+	}
+
+	return corev1.Container{
+		Name:            "policy-recommendation-controller",
+		Image:           pr.image,
+		Env:             envs,
+		SecurityContext: sc,
+		VolumeMounts:    volumeMounts,
+	}
+}
+
+func (pr *policyRecommendationComponent) deployment() *appsv1.Deployment {
+	return &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{Kind: "Deployment", APIVersion: "apps/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      PolicyRecommendationName,
+			Namespace: PolicyRecommendationNamespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: ptr.Int32ToPtr(1),
+			Template: *pr.deploymentPodTemplate(),
+		},
+	}
+}
+
+func (pr *policyRecommendationComponent) deploymentPodTemplate() *corev1.PodTemplateSpec {
+	var ps []corev1.LocalObjectReference
+	for _, x := range pr.cfg.PullSecrets {
+		ps = append(ps, corev1.LocalObjectReference{Name: x.Name})
+	}
+
+	volumes := []corev1.Volume{
+		pr.cfg.TrustedBundle.Volume(),
+		pr.cfg.PolicyRecommendationServerCertSecret.Volume(),
+	}
+
+	container := relasticsearch.ContainerDecorateIndexCreator(
+		relasticsearch.ContainerDecorate(pr.controllerContainer(), pr.cfg.ESClusterConfig.ClusterName(),
+			ElasticsearchPolicyRecommendationUserSecret, pr.cfg.ClusterDomain, rmeta.OSTypeLinux),
+		pr.cfg.ESClusterConfig.Replicas(), pr.cfg.ESClusterConfig.Shards())
+
+	return relasticsearch.DecorateAnnotations(&corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        PolicyRecommendationName,
+			Namespace:   PolicyRecommendationNamespace,
+			Annotations: pr.policyRecommendationAnnotations(),
+		},
+		Spec: corev1.PodSpec{
+			Tolerations:        pr.cfg.Installation.ControlPlaneTolerations,
+			NodeSelector:       pr.cfg.Installation.ControlPlaneNodeSelector,
+			ServiceAccountName: PolicyRecommendationName,
+			ImagePullSecrets:   ps,
+			Containers: []corev1.Container{
+				container,
+			},
+			Volumes: volumes,
+		},
+	}, pr.cfg.ESClusterConfig, pr.cfg.ESSecrets).(*corev1.PodTemplateSpec)
+}
+
+func (pr *policyRecommendationComponent) policyRecommendationAnnotations() map[string]string {
+	annotations := pr.cfg.TrustedBundle.HashAnnotations()
+
+	pr.cfg.TrustedBundle.HashAnnotations()
+	if pr.cfg.PolicyRecommendationServerCertSecret != nil {
+		annotations[pr.cfg.PolicyRecommendationServerCertSecret.HashAnnotationKey()] = pr.cfg.PolicyRecommendationServerCertSecret.HashAnnotationValue()
+	}
+	return annotations
+}
+
+func (pr *policyRecommendationComponent) serviceAccount() client.Object {
+	return &corev1.ServiceAccount{
+		TypeMeta:   metav1.TypeMeta{Kind: "ServiceAccount", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: PolicyRecommendationServiceAccountName, Namespace: PolicyRecommendationNamespace},
+	}
+}
+
+// allowTigeraPolicyForPolicyRecommendation defines an allow-tigera policy for policy recommendation.
+func allowTigeraPolicyForPolicyRecommendation(cfg *PolicyRecommendationConfiguration) *v3.NetworkPolicy {
+	apiserverEntityRule := v3.EntityRule{
```

Comment:
We already have a common definition for this here you can use: https://github.com/tigera/operator/blob/0d5736a65e02ae217bbdeef18492653a71124055/pkg/render/common/networkpolicy/networkpolicy.go#L173-L178

On that note, for egress rules we should be using `KubeAPIServerServiceSelectorEntityRule` instead anyway